diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rom5780-a3-gy.dts b/arch/arm64/boot/dts/rockchip/rk3399-rom5780-a3-gy.dts
index 257c304..4681226 100755
--- a/arch/arm64/boot/dts/rockchip/rk3399-rom5780-a3-gy.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rom5780-a3-gy.dts
@@ -669,7 +669,8 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&rgmii_16ma_pins &rgmii_phy_reset>;
 	enable_phy_delay = <0>;
-	tx_delay = <0x2c>;
+	tx_delay = <0x1c>;
+	//tx_delay = <0x2c>;
 	rx_delay = <0x1e>;
 	status = "okay";
 };
